dashboard/test/models/puzzle_rating_test.rb

```diff
@@ -0,0 +1,38 @@
+require 'test_helper'
+
+class PuzzleRatingTest < ActiveSupport::TestCase
+
+  setup do
+    @student = create :student
+    @script = create :script
+    @level = create :level
+  end
+
+  test "logged-in user can rate unless already rated" do
+    PuzzleRating.stubs(:enabled?).returns(true)
+
+    assert_equal PuzzleRating.can_rate?(@script, @level, @student), true
```

Comment:
this is a bit weird, but the order of arguments for assert_equal is expected first and actual second (doesn't matter when the tests pass but it affects the error message when the test fails).